hal/us_ticker_api.h

```diff
@@ -59,30 +59,31 @@ extern "C" {
  * \defgroup hal_ticker_shared Ticker Hal
  * Low level interface to the ticker of a target
  *
- * # Defined behavior
- * * The function ticker_init is safe to call repeatedly - Verified by test ::ticker_init_test
- * * The function ticker_init resets the internal count and disables the ticker interrupt - Verified by test ::ticker_init_test
- * * Ticker frequency is non-zero and counter is at least 8 bits - Verified by ::ticker_info_test
- * * The ticker rolls over at (1 << bits) and continues counting starting from 0 - Verified by ::ticker_overflow_test
- * * The ticker counts at the specified frequency +- 10% - Verified by ::ticker_frequency_test
- * * The ticker increments by 1 each tick - Verified by ::ticker_increment_test
+ * # Defined behaviour
+ * * The function ticker_init is safe to call repeatedly - Verified by test ticker_init_test()
```

Comment:
Without the :: is doxygen still able to pick up these function as links?

Reply:
I'm not sure. As you can see in this conversation I had problems with doxygen and travis. 
According to info from doxygen manual the solution was to use `func()` instead of `::func` format.

Reply:
I build the doxygen locally and I can confirm that the link no longer works. This should be reverted back to the :: version. If you need to build the doxygen locally to help debug travis you can run ```doxygen doxyfile_options``` if you have doxygen installed and on your path.

Reply:
The `::` wasn't working in the first place, the doxygen was failing to build it both locally and in travis.

Reply:
According to documentation provided by @bulislaw both options should work, but we have the following cases:
- when `::` option is used Travis fails with reference could not be resolved error on links to test functions.
- when `()` option is used Travis does not report any error, but links in documentation are not created (probably in this case doxygen does not try to create links - maybe some option must be set?). 

This means that we should use `::` and find the reason of this error. I have modified back style from `()` to `::`  (note: this method was used on feature RTC branch and it worked).

I have checked it again locally and the problem is with the test functions which are not visible in the generated document. That is why references for these functions can not be resolved.
The following groups are empty: `hal_ticker_tests`, `hal_us_ticker_tests`, `hal_lp_ticker_tests`. Test headers provide doxygen command `\addtogroup  hal_(xx_)ticker_tests` and `TEST` directory is not excluded in doxygen options. We are missing something, but I can't see what is wrong.

Additionally when I try to run locally `doxygen doxyfile_options` command I get `Segmentation fault`, but the document is generated.